pkgs/applications/science/misc/root/default.nix

```
@@ -125,6 +127,8 @@ stdenv.mkDerivation rec {
     # Eliminate impure reference to /System/Library/PrivateFrameworks
     substituteInPlace core/CMakeLists.txt \
       --replace "-F/System/Library/PrivateFrameworks" ""
+  '' + lib.optionalString (lib.versionAtLeast stdenv.hostPlatform.darwinMinVersion "11" && stdenv.isDarwin) ''
```

Though `stdenv.hostPlatform.darwinMinVersion` still evaluates to a string, it sounds more sensible to check `stdenv.isDarwin` first and then Darwin-specific conditions.

```suggestion
  '' + lib.optionalString (stdenv.isDarwin && lib.versionAtLeast stdenv.hostPlatform.darwinMinVersion "11") ''
```

---

Done.